eng/restore-toolset.ps1

```diff
@@ -53,6 +53,27 @@ set PATH=$env:DOTNET_INSTALL_DIR;%PATH%
 set NUGET_PACKAGES=$env:NUGET_PACKAGES
 
 DOSKEY killdotnet=taskkill /F /IM dotnet.exe /T ^& taskkill /F /IM VSTest.Console.exe /T ^& taskkill /F /IM msbuild.exe /T
+"@
+
+  Out-File -FilePath $scriptPath -InputObject $scriptContents -Encoding ASCII
+
+  Create-Directory $ArtifactsDir
+  $scriptPath = Join-Path $ArtifactsDir "sdk-build-env.ps1"
+  $scriptContents = @"
+`$host.ui.RawUI.WindowTitle` = "SDK Build ($RepoRoot)"
+`$env:DOTNET_MULTILEVEL_LOOKUP`=0
+
+`$env:DOTNET_ROOT`="$env:DOTNET_INSTALL_DIR"
+`$env:DOTNET_MSBUILD_SDK_RESOLVER_CLI_DIR`="$env:DOTNET_INSTALL_DIR"
+
+`$env:PATH`="$env:DOTNET_INSTALL_DIR;$env:PATH"
+`$env:NUGET_PACKAGES`="$env:NUGET_PACKAGES"
```

Comment:
Why does this backquote-escape the equals signs?

Reply:
the equals are literal strings, the backquotes were added so that at string-interpolation time the values of the shell executing the `restore-toolset.ps1` script don't get interpolated. make sense?

Reply:
here's what the rendered ps1 looks like on my machine:

```powershell
$host.ui.RawUI.WindowTitle = "SDK Build (C:\Users\chethusk\OSS\dotnet-sdk\)"
$env:DOTNET_MULTILEVEL_LOOKUP=0

$env:DOTNET_ROOT="C:\Users\chethusk\OSS\dotnet-sdk\.dotnet"
$env:DOTNET_MSBUILD_SDK_RESOLVER_CLI_DIR="C:\Users\chethusk\OSS\dotnet-sdk\.dotnet"

$env:PATH="C:\Users\chethusk\OSS\dotnet-sdk\.dotnet;C:\Users\chethusk\OSS\dotnet-sdk\.dotnet;C:\Users\chethusk\OSS\dotnet-sdk\.dotnet;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\\Extensions\Microsoft\IntelliCode\CLI;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin\Roslyn;C:\Program Files\Microsoft Visual Studio\2022\Preview\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools;C:\Program Files\Microsoft Visual Studio\2022\Preview\\MSBuild\Current\Bin\amd64;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\Tools\;C:\Python310\Scripts\;C:\Python310\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\dotnet-core-uninstall\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Go\bin;C:\Users\chethusk\AppData\Local\Microsoft\WindowsApps;C:\Users\chethusk\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\chethusk\.dotnet\tools;C:\Users\chethusk\Tools;C:\Users\chethusk\AppData\Roaming\npm;C:\Users\chethusk\OSS\dotnet-sdk\artifacts\tmp\Debug\.dotnet\tools;C:\Users\chethusk\go\bin;C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\VC\Linux\bin\ConnectionManagerExe"
$env:NUGET_PACKAGES="C:\Users\chethusk\.nuget\packages\"

function killdotnet {
  taskkill /F /IM dotnet.exe /T 
  taskkill /F /IM VSTest.Console.exe /T 
  taskkill /F /IM msbuild.exe /T
}
```

without the backticks, the left-hand sides of the equals are evaluated while making the script.

Reply:
I expect the backquote before the dollar sign would suffix for that; no need to have another for the equals sign.

Reply:
you're exactly right - I though the ticks worked like encapsulation, but instead it just escapes the offending character. Thanks!